capstone/src/main/java/com/google/sps/data/RecommenderAlgorithm.java

```diff
@@ -11,49 +13,158 @@
 import com.google.appengine.api.datastore.Query.FilterPredicate;
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.Entity;
-import java.util.HashMap;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.FetchOptions;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.TreeMap;
+import java.util.Map;
 import java.lang.Iterable;
+import java.util.Collections;
+import java.util.Comparator;
+import java.lang.Math;
  
 public class RecommenderAlgorithm {
- 
-    public static ArrayList<Room> recommendRooms(User user) {
+
+    // TODO: Will add the proportions in the next PR when I know exact percentages
+
+    public static List<Key> recommendRooms(User user, Integer amount) {
+        RecommenderAlgorithm recommender = null;
+        recommender.normalizeAllEmbeddedEntities();
+
+        Map<Key, Double> distances = recommender.getDistanceMaps(user, user.getEmbeddedTags());
+        Map sorted = sortByValues(distances);
```

Comment:
Upcasting is dangerous here, because you actually care a lot about the specific kind of map it is. Since you are relying on the order of the map for this function to work, it is not enough for this to just be the Map type. I may see this and be like "oh I can refactor this, I have another way to get a map of all the distances, I can just delete replace this", but actually, you need sorted distances, so this should explicitly be of the TreeMap type

Reply:
Although... in reality, you dont need a map at all. Since all you are getting is the key, this should just return a list of Keys

Reply:
good point, will change the return type

Reply:
only reason I used TreeMaps is so that I could sort the values. If I just returned a list of keys I could do the sorting in the method I guess?